vsintegration/src/FSharp.Editor/CodeFixes/CodeFixHelpers.fs

```diff
@@ -106,3 +108,57 @@ module internal CodeFixExtensions =
                 let! sourceText = ctx.GetSourceTextAsync()
                 return RoslynHelpers.TextSpanToFSharpRange(ctx.Document.FilePath, ctx.Span, sourceText)
             }
+
+// This cannot be an extension on the code fix context
+// because the underlying GetFixAllProvider method doesn't take the context in.
+#nowarn "3511" // state machine not statically compilable
+
+[<AutoOpen>]
+module IFSharpCodeFixProviderExtensions =
+    type IFSharpCodeFixProvider with
+
+        member provider.RegisterFsharpFixAll() =
+            FixAllProvider.Create(fun fixAllCtx doc allDiagnostics ->
+                cancellableTask {
+                    let sw = Stopwatch.StartNew()
+
+                    let! token = CancellableTask.getCurrentCancellationToken ()
+                    let! sourceText = doc.GetTextAsync token
+
+                    // this is not used anywhere, it's just needed to create the context
+                    let action =
+                        Action<CodeActions.CodeAction, ImmutableArray<Diagnostic>>(fun _ _ -> ())
+
+                    let! codeFixOpts =
+                        allDiagnostics
+                        // The distiction is to avoid collisions of compiler and analyzer diags
+                        // See: https://github.com/dotnet/fsharp/issues/15620
```

Comment:
Why not changing the (artificial) diagnostics ID produced by the Analyzer instead then?

Reply:
Or, alternatively, disable the dedicated analyzer in case of waron being defined, to save resources.
If, under warnon, the compiler does it already, we can save the work of doing the work again by the analyzer.

Reply:
I don't really get the part of not being in control- "we" for Roslyn external access is the same "we" as making changes in the repo.
The fact that something is defined in another repo should not be an excuse for not changing it and instead doing workarounds.

Reply:
Yes - but can we have that as a followup? This will take some time - picking the new diag, pushing it to roslyn, explaining it to roslyn and consuming it in F#. I don't mind doing this, I just want to meanwhile keep working on code fixes as this PR also contains some framework/interface changes I want to build on top of.

Or - I can revert this change so that we keep living with this bug for a few more weeks.